Backport of upstream bf85f8a (Workaround fix for systems without MPFR and GMP).
Fixes make check failures when built without MPFR: unassigned elems print as "0".
https://lists.gnu.org/archive/html/bug-gawk/2026-07/msg00027.html
https://git.savannah.gnu.org/cgit/gawk.git/commit/?id=bf85f8a3175af703597082d4c7e0abc2066a44d3

--- a/awk.h
+++ b/awk.h
@@ -406,17 +406,24 @@
 		} nodep;
 
 		struct {
-#ifdef HAVE_MPFR
 			union {
 				AWKNUM fltnum;
+#ifdef HAVE_MPFR
 				mpfr_t mpnum;
 				mpz_t mpi;
-			} nm;
-			int rndmode;
 #else
-			AWKNUM fltnum;
-			int for_alignment_only;	// especially on 32-bit
-#endif
+				// 7/2026:
+				// This is a workaround for systems that build
+				// gawk without MPFR and GMP.  The NODE struct
+				// desperately needs to be refactored.
+#if SIZEOF_VOID_P == 4
+				char alignment[28];
+#else	// SIZEOF_VOID_P != 4
+				char alignment[48];
+#endif	// SIZEOF_VOID_P != 4
+#endif	// HAVE_MPFR
+			} nm;
+			int rndmode;	// only used for MPFR.
 			char *sp;
 			size_t slen;
 			int idx;
@@ -561,10 +568,8 @@
 #ifdef HAVE_MPFR
 #define mpg_numbr	sub.val.nm.mpnum
 #define mpg_i		sub.val.nm.mpi
-#define numbr		sub.val.nm.fltnum
-#else
-#define numbr		sub.val.fltnum
 #endif
+#define numbr		sub.val.nm.fltnum
 #define typed_re	sub.val.typre
 
 /*
